src/util/std_code.h

```diff
@@ -296,14 +297,14 @@ class code_assignt:public codet
 
   static void validate(
     const codet &code,
-    const namespacet &,
+    const namespacet &ns,
     const validation_modet vm = validation_modet::INVARIANT)
   {
     check(code, vm);
 
     DATA_CHECK(
       vm,
-      code.op0().type() == code.op1().type(),
+      base_type_eq(code.op0().type(), code.op1().type(), ns),
```

Comment:
To be debated with @kroening as this appears to work against the effort going on in other PRs.

Reply:
Fair enough if we want the strict same-level-of-indirection condition in the future, but we certainly don't even try to do that at present.

Reply:
Can this commit be dropped from the PR? I understand that this would mean it doesn't work for Java at the moment, but then we can get all the other improvements in the meantime.

Reply:
Well, my intent was to use this as the means of testing the other improvements, by showing particular programs begin to satisfy this constraint when the fixes are applied, so I'd much prefer to have this statement of our current intended guarantee available, then we can chase after a future attempted standard later?